### PR TITLE
Show Toast using JavaScript function in WebView

### PR DESCRIPTION
**Pull request by infinyte7**
## Purpose / Description
As sometime it needed to toast in AnkiDroid. This will create native toast of message passed by card developer
This is continuation of #6307

## Fixes
Fixes _Link to the issues._

## Approach
Added signal for calling function
In card.js
```
ankiShowToast(msg)
```

## How Has This Been Tested?
Tested on emulator,
The function accept string for showing toast message
In front side of card template add it in ```<script>``` tag
```
ankiShowToast("Showing native toast using javascript");
```

## Learning (optional, can help others)
https://developer.android.com/guide/topics/ui/notifiers/toasts

## Checklist
_Please, go through these checks before submitting the PR._

- [ x ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ x ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ x ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ x ] You have commented your code, particularly in hard-to-understand areas
- [ x ] You have performed a self-review of your own code

### CODE REVIEW

**Comment by infinyte7** [comment]
This is not much required as using CSS it can be made. But it will be good if there because using one line by passing message can show toast in native way.

File: AnkiDroid/src/main/assets/scripts/card.js [hunk 1]
```diff
@@ -125,6 +125,11 @@ function ankiToggleFlag(flag) {
     }
 }
 
+// Show toast using js
+function ankiShowToast(message) {
+    window.location.href = "signal:anki_show_toast" + message;
```
Comment (david-allison-1) [inline AnkiDroid/src/main/assets/scripts/card.js:130]:
```suggestion
    window.location.href = "signal:anki_show_toast:" + message;
```

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 2]
```diff
@@ -3110,6 +3110,13 @@ private boolean filterUrl(String url) {
                 }
             }
 
+            // Show toast using JS
+            if (url.startsWith("signal:anki_show_toast")) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3114]:
```suggestion
            if (url.startsWith("signal:anki_show_toast:")) {
```

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 3]
```diff
@@ -3110,6 +3110,13 @@ private boolean filterUrl(String url) {
...
+                String msg = url.replaceFirst("signal:anki_show_toast","");
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3115]:
```suggestion
                String msg = url.replaceFirst("signal:anki_show_toast:", "");
```

**Comment by infinyte7** [comment]
I will update it for other language also.

**Comment by infinyte7** [comment]
Now working for other languages.
 
toast_demo

**Comment by david-allison-1** [comment]
I believe this will break if you enter a single percentage: `100%` for example. Could you test this against that case?

**Comment by infinyte7** [comment]
Yes, it break when single % supplied,
</img>

**Comment by infinyte7** [comment]
Now working

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 4]
```diff
@@ -3110,6 +3110,14 @@ private boolean filterUrl(String url) {
                 }
             }
 
+            // Show toast using JS
+            if (url.startsWith("signal:anki_show_toast:")) {
+                String msg = url.replaceFirst("signal:anki_show_toast:", "");
+                String msgDecode = Uri.decode(msg);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:3116]: This needs to be in a try.. catch as the user can manually trigger a signal with bad input.

Comment (infinyte7) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java]: Updated

**Comment by infinyte7** [comment]
Is it okay to be merged?

**Pull request merged**